python/phylanx/ast/physl.py

```diff
@@ -422,27 +436,19 @@ def call(self, args):
         if not self.is_compiled:
             if "compiler_state" in self.kwargs:
                 PhySL.compiler_state = self.kwargs['compiler_state']
-            # the static method compiler_state is constructed only once
             elif PhySL.compiler_state is None:
                 PhySL.compiler_state = compiler_state()
 
             phylanx.execution_tree.compile(
                 self.file_name, self.__src__, PhySL.compiler_state)
             self.is_compiled = True
 
-        result = phylanx.execution_tree.eval(
-            self.file_name, func_name, PhySL.compiler_state, *args)
+        self.args = args
+        return self
 
```

Comment:
This will fail if `lazy()` is called more than once to retrieve differently bound code objects before `eval()` is invoked on those. I think `lazy()` should return an object that exposes an `eval()` function and stores the `args` array in order to pass the correct arguments once invoked.

Reply:
This was fixed by 801e117